pipelines/release/release.yaml

```diff
@@ -444,6 +456,61 @@ jobs:
             EOF
             echo "merging with cluster values..."
             spruce merge ./platform/charts/gsp-cluster/values.yaml ./overrides.yaml | tee -a cluster-values/values.yaml
+  - task: pick-platform-version
+    image: concourse-task-toolbox
+    config:
+      platform: linux
+      inputs:
+      - name: full-gsp
+      - name: platform
+      - name: service-operator-source
+      - name: concourse-task-toolbox-source
+      - name: concourse-operator-source
+      - name: concourse-github-resource-source
+      - name: concourse-harbor-resource-source
+      outputs:
+      - name: platform-version
+      run:
+        path: /usr/bin/python3
+        args:
+        - -c
+        - |
+          import os
+          import subprocess
+          import sys
+
+          os.mkdirs("platform-version", exist_ok=True)
+
+          print("Picking platform version...")
+          partial_repos = [
+              "platform",
+              "service-operator-source",
+              "concourse-task-toolbox-source",
+              "concourse-operator-source",
+              "concourse-github-resource-source",
+              "concourse-harbor-resource-source"
+          ]
+          hashes = set(map(lambda r: open(f"{r}/.git/ref").read(), partial_repos))
+
+          proc = subprocess.Popen(
+              ['git', 'log', '--format=%H'],
+              env={'GIT_DIR': 'full-gsp/.git'},
+              stdout=subprocess.PIPE
+          )
+          while True:
+              line = proc.stdout.readline().decode('ascii')
+              if not line:
+                  sys.stderr.write("Did not get line when running git-log through full-gsp\n")
+                  sys.exit(1)
+              if line in hashes:
+                  proc.terminate()
+                  break
+
+          line = line.strip()
```

Comment:
If we don't find a match `line` will be the last entry searched. It might be an unlikely scenario, but I'd like us to catch it if we can with a check and an explicit error message.

Reply:
If we don't find a match `sys.exit(1)` will have been called.